rust/src/server.rs

```diff
@@ -259,6 +261,15 @@ fn initialize(
     );
     state.phase = IndexerPhase::Watching;
 
+    // flush/compress database
+    let random_number = rand::thread_rng().gen_range(1..=100);
+    let temp_checkpoint_dir = std::env::temp_dir().join(format!("snapshot_{random_number}"));
```

Comment:
Need to make sure that the temporary directory used is on the same volume as the database. Best to create an "adjacent" file in the filesystem, there. Otherwise, the temporary directory will not likely have enough space.

Reply:
Also, if it's not on the same device, the checkpoint will copy the files over rather than create hard-links